doc/icache.rst

```diff
@@ -219,12 +219,16 @@ If that happens, the cache drops ``instr_req_o`` and stops making any further re
 Note that it is possible for ``instr_gnt_i`` and ``instr_pmp_err_i`` to be high on the same cycle.
 In that case, the error signal takes precedence.
 
-Fetched instructions are returned to the core using ports ``ready_i``, ``valid_o``, ``rdata_o``, ``addr_o`` and ``err_o``.
-This interface uses ready/valid handshaking in the usual way (a transaction is signalled by ready and valid being high; if valid goes high, it will remain high and the other output signals will remain stable until the transaction goes through).
-The one exception is if ``branch_i`` is asserted, which will cause ``valid_o`` to de-assert.
+Fetched instructions are returned to the core using ports ``ready_i``, ``valid_o``, ``rdata_o``, ``addr_o``, ``err_o`` and ``err_plus2_o``.
+This interface uses a form of ready/valid handshaking.
+A transaction is signalled by ready and valid being high.
+If valid goes high, it will remain high and the other output signals will remain stable until the transaction goes through or is cancelled by ``branch_i`` being asserted.
+There is no constraint on the behaviour of ``ready_i``.
+
 The 32-bit wide ``rdata_o`` signal contains instruction data fetched from ``addr_o``.
-If there is a compressed instruction in the lower 16 bits, the upper 16 bits are unconstrained.
-This allows fetching an instruction from the top 16 bits of a memory, for example.
+An instruction is either 16 or 32 bits wide (called *compressed* or *uncompressed*, respectively).
+The width of an instruction can be calculated from its bottom two bits: an instruction is uncompressed if the equal ``2'b11`` and compressed otherwise.
```

Comment:
the -> they

Reply:
Thanks! :-)

Reply:
Incidentally, I don't think I was clear in the commit message: the original text was also mine, I think. I might not have been quite so certain that things were getting better if it was someone else's prose!